solver/tests/basic.rs

```diff
@@ -0,0 +1,133 @@
+use std::collections::HashMap;
+
+use leafsolver::{
+    Ast, AstAndVars, AstNode, BVNode, Config, Constraint, ConstraintKind, Context, SolveResult,
+    Z3Solver, ast,
+};
+
+#[test]
+fn test_constraint_creation() {
+    let constraint = Constraint::equality("x", 5);
+
+    assert_eq!(constraint.discr, "x");
+    match constraint.kind {
+        ConstraintKind::OneOf(ref values) => {
+            assert_eq!(values.len(), 1);
+            assert_eq!(values[0], 5);
+        }
+        _ => panic!("Expected OneOf constraint"),
+    }
+}
+
+#[test]
+fn test_constraint_negation() {
+    let constraint = Constraint::equality("x", 5);
+    let negated = constraint.not();
+
+    match negated.kind {
+        ConstraintKind::NoneOf(ref values) => {
+            assert_eq!(values.len(), 1);
+            assert_eq!(values[0], 5);
+        }
+        _ => panic!("Expected NoneOf constraint after negation"),
+    }
+}
+
+#[test]
+fn test_constraint_kind_operations() {
```

Comment:
Thanks for writing the tests. I see some of them are focused on `common`'s data structures. Why not moving them there next to their definitions?